### ISSUE TITLE
Infinite inertia

### ISSUE THREAD

**Issue by Leakbang** [Btan2/Q_Move#1]
At the moment, the player character has infinite force and because of this, interactions with rigidbodies go horribly wrong. 
In Half-Life 2, the player had a specific weight that restricted how much force it inflicted on physics objects and it allowed the player to push and stand on objects. The player would naturally slow down while pushing an object based on it's weight and when the object could not be pushed any further, the player momentum stops. This also allowed the player to stand on objects (i.e. Crates and boxes).
At the moment, the player has infinite weight and force so standing on an object would just crush the object into the floor and pushing objects don't slow the player down at all.

**Comment by Leakbang** [comment]
Changing the player type from a Kinematicbody to a Rigidbody can fix this issue. I believe that having the player as a rigidbody has many more additional benefits. It will result in the player being able to interact with the world environment in a very robust and realistic manner. 
The issue of the player sliding down ramps could also be automatically fixed by switching to a Rigidbody base.

**Comment by Btan2** [comment]
I haven't taken rigidbodies into consideration yet, just static collision events for the time being. I'll give rigidbodies a try when I get time. Thanks Leakbang.

**Comment by Btan2** [comment]
So from a quick internet search I managed to solve the infinite inertia issue for the time being:
https://kidscancode.org/godot_recipes/physics/kinematic_to_rigidbody/
Infinite inertia can be disabled in the "move_and_collide" function and then you can simply apply an impulse to the rigidbody you're colliding with to push it. It works pretty well for now, although jumping up and down on a rigidbody box can push it in unexpected directions. Perhaps this can be solved by not applying an impulse if the rigidbody is grounded and the impulse vector is pushing it downwards. But like you said, switching to a Rigidbody controller might automatically solve these problems.

**Comment by Leakbang** [comment]
Yes that solves the infinite inertia issue but the kinematicbody-rigidbody interactions will still be an issue. Applying impulse on collision makes it feel like that the player is kicking a ball. And for instance if the player wants to move a heavy box, it just gets launched and the player loses no velocity. But if the player is a Rigidbody and is being moved around by physical forces (probably by using`add_central_force`), then a box on the player's way will slow the player down based on how heavy the box actually is. And if the box is too heavy, then the player can't move it at all. What's so great about this, is that it lays down a very robust and flexible groundwork that can be expanded upon.
